python/pyarrow/tests/test_dataset.py

```diff
@@ -1587,33 +1587,54 @@ def test_open_dataset_non_existing_file():
 
 @pytest.mark.parquet
 @pytest.mark.parametrize('partitioning', ["directory", "hive"])
+@pytest.mark.parametrize('null_fallback', ['xyz', None])
```

Comment:
What does `null_fallback=None` mean? (based on the docstring above it seems it can only be a string?)

Reply:
In that case it does not pass in anything when it creates the partitioning and ensures that it defaults to the correct default value.